lib/impure/rdstdin.nim

```diff
@@ -100,12 +100,17 @@ when defined(Windows):
     stdout.write "\n"
 
 else:
-  import linenoise, termios
+  import linenoise, termios, posix
 
   proc readLineFromStdin*(prompt: string): TaintedString {.
                           tags: [ReadIOEffect, WriteIOEffect].} =
+    errno = 0
```

Comment:
Errno can be resolved to a function call on some targets, it should not be set directly. Also 'errno' is for posix, not for user level code.